src/vimish/commands/hilite.js

```diff
@@ -1,61 +1,104 @@
-      pattern = pattern.join(" ")
-      if (!pattern || pattern.length == 0) {
-        throw new Error(
-          ":hilite add <pattern> <group> was missing pattern"
-        )
-      }
-      if (!group) {
-        throw new Error(":hilite add <pattern> <group> was missing group")
-      }
-      return Hilites.add_pattern(group, pattern)
-    }
+/**
+ * Add a pattern to the hilite database.
+ * @param {Array<string>} rest Arguments to the command
+ * @param {boolean} sudo Flag indicating if this was invoked with sudo
+ */
+function add_pattern(rest, _) {
+  let [group, ...pattern] = rest
+  pattern = pattern.join(" ")
+  if (!pattern || pattern.length == 0) {
+    throw new Error(":hilite add <pattern> <group> was missing pattern")
+  }
+  if (!group) {
+    throw new Error(":hilite add <pattern> <group> was missing group")
+  }
+  return Hilites.add_pattern(group, pattern)
+}
 
-    if (
-      (opts.sub_command == "rm" || opts.sub_command == "remove") &&
-      sudo
-    ) {
-      throw new Error(":hilite remove not implemented yet")
-      // todo: sudo ?
-    }
+/**
+ * Add a group to the database
+ * @param {Array<string>} rest Arguments to the command
+ * @param {boolean} sudo flag indicating if this was invoked with sudo
+ */
+function add_group(rest, _) {
+  const [group, ...rules] = rest
 
-    if (opts.sub_command == "rm" || opts.sub_command == "remove") {
-      throw new Error(":sudo is required for :hilite rm")
-    }
+  if (!rules.length) {
+    throw new Error(
+      `:hilite group <group> ...<rules> requires at least 1 rule`
+    )
+  }
 
-    if (opts.sub_command == "reload") {
-      return Hilites.reload()
-    }
+  return Hilites.add_group(
+    group,
+    rules
+      .map((rule) => rule.split("="))
+      .reduce(
+        (acc, [rule, value]) => Object.assign(acc, { [rule]: value }),
+        {}
+      )
+  )
+}
 
-    if (opts.sub_command == "group") {
-      const [group, ...rules] = rest
+/**
+ * Remove a group or pattern from the hilite database.
+ * @param {Array<string>} rest Arguments to the command
+ * @param {boolean} sudo Flag indicating if this was invoked with sudo
+ */
+function remove_command(rest, sudo) {
+  if (!sudo) {
+    throw new Error(":sudo is required for :hilite rm")
+  }
+  let [rmClass, ...pattern] = rest
+  if (!pattern || pattern.length == 0) {
+    throw new Error(
+      `:sudo :hilite remove ${rmClass} requires an extra argument.`
+    )
+  }
+  switch (rmClass) {
+    case "group":
+      if (
+        Object.values(Hilites.Patterns).some((elem) => elem == pattern)
+      ) {
+        if (pattern[0] != "confirm") {
+          throw new Error(
+            `${pattern} is still in use. Please use :sudo :hilist remove group confirm <pattern> to force removal.`
+          )
+        }
+        pattern = pattern.slice(1)
+      }
+      Hilites.remove_group(pattern.join(" "))
+      break
+    case "pattern":
+      Hilites.remove_pattern(pattern.join(" "))
```

Comment:
this should be refactored to use `return Hilites....` instead of dangling breaks